File: speakinsights-v3/backend/app/schemas/transcription.py
"""
SpeakInsights v3 — Transcription Schemas
"""

from pydantic import BaseModel
from datetime import datetime
from uuid import UUID
from typing import Optional


class TranscriptSegmentResponse(BaseModel):
    """Schema for a single transcript segment."""
    id: UUID
    meeting_id: UUID
    speaker_name: str
    text: str
    language: Optional[str] = None
    start_time: float
    end_time: float
    confidence: Optional[float] = None
    sentiment_score: Optional[float] = None
    sentiment_label: Optional[str] = None
    word_count: Optional[int] = None
    source: str = "live"
    created_at: datetime

    model_config = {"from_attributes": True}


class SpeakerTimeline(BaseModel):
    """Transcript segments grouped by speaker."""
    speaker_name: str
    segments: list[TranscriptSegmentResponse]
    total_duration: float = 0.0
    total_words: int = 0


class TranscriptTimelineResponse(BaseModel):
    """Timeline view of transcript grouped by speaker."""
    meeting_id: UUID
    speakers: list[SpeakerTimeline]
    total_segments: int
    total_duration: float


class FullTranscriptResponse(BaseModel):
    """Full transcript as ordered segments."""
    meeting_id: UUID
    segments: list[TranscriptSegmentResponse]
    total_segments: int
    total_duration: float
    languages: list[str] = []


class TranscriptSegmentCreate(BaseModel):
    """Schema for creating a transcript segment (from WhisperX)."""
    speaker_name: str
    text: str
    language: Optional[str] = None
    start_time: float
    end_time: float
    confidence: Optional[float] = None
    source: str = "live"

File: speakinsights-v3/frontend/src/types/transcription.ts
export interface Word {
  word: string;
  start: number;
  end: number;
  score?: number;
}

export interface Speaker {
  name: string;
  id: string;
  color?: string;
}

export interface TranscriptSegment {
  id: string;
  meeting_id: string;
  speaker_name: string;
  text: string;
  start_time: number;
  end_time: number;
  words?: Word[];
  sentiment_score?: number;
  sentiment_label?: string;
  is_live?: boolean;
  created_at: string;
}

export interface TranscriptTimeline {
  segments: TranscriptSegment[];
  speakers: Speaker[];
  duration: number;
}

export interface TranscriptSearchResult {
  segment: TranscriptSegment;
  score: number;
  highlight: string;
}
